project doesn't build on macos

i have a macbookpro M1 from 2020, macos version is "monterey 12.3.1 (21E258)". i tried the two types of build, both failed.
here's what i got:

build type 1 (single binary)
```
mkdir build
cd build
cmake ..
make

results in:
[ build ]                                                                             ⍉
> pwd
/Users/klaus/Sources/dolphin/build

[ build ]
> make
[  0%] Built target cpp-optparse
[  1%] Built target imgui
[  4%] Built target glslang
[  4%] Performing update step for 'MoltenVK'
[  4%] Performing patch step for 'MoltenVK'
HEAD is now at b3ef160c Merge pull request #1470 from billhollings/sdk-1.2.198
[  4%] Performing configure step for 'MoltenVK'
CMake Error at /Users/klaus/Sources/dolphin/build/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp/MoltenVK-configure-Release.cmake:37 (message):
  Command failed: 127

   '/Users/klaus/Sources/dolphin/Externals/MoltenVK/configure.sh' '/Users/klaus/Sources/dolphin/build/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp' '/Users/klaus/Sources/dolphin/build/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK' 'v1.1.6'

  See also

    /Users/klaus/Sources/dolphin/build/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp/MoltenVK-configure-*.log


-- stdout output is:

========== Retrieving MoltenVK dependencies into External ==========

========== Cereal ==========

cereal repo: https://github.com/USCiLab/cereal.git
cereal revision: 51cbda5f30e56c801c07fe3d3aba5d7fb9e6cca4
Fetching origin

========== Vulkan-Headers ==========

Vulkan-Headers repo: https://github.com/KhronosGroup/Vulkan-Headers.git
Vulkan-Headers revision: 8c1c27d5a9b9de8a17f500053bd08c7ca6bba19c
Fetching origin

========== SPIRV-Cross ==========

SPIRV-Cross repo: https://github.com/KhronosGroup/SPIRV-Cross.git
SPIRV-Cross revision: 7c3cb0b12c9965497b08403c82ac1b82846fa7be
Fetching origin

========== glslang and SPIRV-Tools ==========

glslang repo: https://github.com/KhronosGroup/glslang.git
glslang revision: c9706bdda0ac22b9856f1aa8261e5b9e15cd20c5
Fetching origin

-- stderr output is:
cat: /Users/klaus/Sources/dolphin/build/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp/MoltenVK-last-version.txt: No such file or directory
HEAD is now at 51cbda5f Merge branch 'develop' for release 1.2.2
HEAD is now at 8c1c27d Update for Vulkan-Docs 1.2.198
HEAD is now at 7c3cb0b1 Merge pull request #1810 from billhollings/early-frag-tests-depth-out
HEAD is now at c9706bdd Merge pull request #2817 from greg-lunarg/sdk0
env: python: No such file or directory

CMake Error at /Users/klaus/Sources/dolphin/build/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp/MoltenVK-configure-Release.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp/MoltenVK-configure] Error 1
make[1]: *** [Externals/MoltenVK/CMakeFiles/MoltenVK.dir/all] Error 2
make: *** [all] Error 2

[ build ]                                                                             ⍉
>
```

build type 2 (universal)
```
[ 61%] Building CXX object Source/Core/Core/CMakeFiles/core.dir/PowerPC/Jit64/Jit_Branch.cpp.o
[ 61%] Building CXX object Source/Core/Core/CMakeFiles/core.dir/PowerPC/Jit64/Jit_LoadStorePaired.cpp.o
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:124:11: warning: 10 enumeration values not handled in switch: 'INVALID_GAME', 'MENU', 'MENU_PRIME_1'... [-Wswitch]
  switch (game) {
          ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:124:11: note: add missing switch cases
  switch (game) {
          ^
[ 61%] Building CXX object Source/Core/Core/CMakeFiles/core.dir/PowerPC/Jit64/Jit_Paired.cpp.o
[ 62%] Building CXX object Source/Core/Core/CMakeFiles/core.dir/PowerPC/Jit64/Jit_SystemRegisters.cpp.o
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:380:20: error: 'path' is unavailable: introduced in macOS 10.15
  std::filesystem::path search_name(str);
                   ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:909:24: note: 'path' has been explicitly marked unavailable here
class _LIBCPP_TYPE_VIS path {
                       ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:380:25: error: 'path' is unavailable: introduced in macOS 10.15
  std::filesystem::path search_name(str);
                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:948:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:380:25: error: '~path' is unavailable: introduced in macOS 10.15
  std::filesystem::path search_name(str);
                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:382:19: error: 'is_absolute' is unavailable: introduced in macOS 10.15
  if (search_name.is_absolute() && std::filesystem::exists(str)) {
                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:1406:34: note: 'is_absolute' has been explicitly marked unavailable here
  _LIBCPP_INLINE_VISIBILITY bool is_absolute() const {
                                 ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:382:53: error: 'exists' is unavailable: introduced in macOS 10.15
  if (search_name.is_absolute() && std::filesystem::exists(str)) {
                                                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:1986:39: note: 'exists' has been explicitly marked unavailable here
inline _LIBCPP_INLINE_VISIBILITY bool exists(const path& __p) {
                                      ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:382:60: error: 'path' is unavailable: introduced in macOS 10.15
  if (search_name.is_absolute() && std::filesystem::exists(str)) {
                                                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:948:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:382:60: error: '~path' is unavailable: introduced in macOS 10.15
  if (search_name.is_absolute() && std::filesystem::exists(str)) {
                                                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:35: error: 'absolute' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:1846:39: note: 'absolute' has been explicitly marked unavailable here
inline _LIBCPP_INLINE_VISIBILITY path absolute(const path& __p) {
                                      ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:44: error: 'path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:948:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:44: error: '~path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                           ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:18: error: '~path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                 ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:54: error: 'parent_path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:1366:34: note: 'parent_path' has been explicitly marked unavailable here
  _LIBCPP_INLINE_VISIBILITY path parent_path() const {
                                 ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:54: error: '~path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                                     ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:71: error: 'path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                                                      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:948:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:71: error: '~path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                                                      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:69: error: 'operator/' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                                                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:1500:41: note: 'operator/' has been explicitly marked unavailable here
  friend _LIBCPP_INLINE_VISIBILITY path operator/(const path& __lhs,
                                        ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:69: error: '~path' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
                                                                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:968:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:385:15: error: 'operator=' is unavailable: introduced in macOS 10.15
  search_name = (std::filesystem::absolute(rel_file).parent_path()) / str;
              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:978:9: note: 'operator=' has been explicitly marked unavailable here
  path& operator=(path&& __p) noexcept {
        ^
/Users/klaus/Sources/dolphin/Source/Core/Core/PrimeHack/Mods/ElfModLoader.cpp:387:24: error: 'exists' is unavailable: introduced in macOS 10.15
  if (std::filesystem::exists(search_name)) {
                       ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/filesystem:1986:39: note: 'exists' has been explicitly marked unavailable here
inline _LIBCPP_INLINE_VISIBILITY bool exists(const path& __p) {
                                      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
1 warning and 20 errors generated.
make[2]: *** [Source/Core/Core/CMakeFiles/core.dir/PrimeHack/Mods/ElfModLoader.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Source/Core/Core/CMakeFiles/core.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- MoltenVK build command succeeded.  See also /Users/klaus/Sources/dolphin/build/x86_64/Externals/MoltenVK/MoltenVK-prefix/src/MoltenVK-stamp/MoltenVK-build-*.log
[ 62%] No install step for 'MoltenVK'
[ 62%] Completed 'MoltenVK'
[ 62%] Built target MoltenVK
make: *** [all] Error 2
Traceback (most recent call last):
  File "/Users/klaus/Sources/dolphin/build/../BuildMacOSUniversalBinary.py", line 367, in <module>
    build(conf)
  File "/Users/klaus/Sources/dolphin/build/../BuildMacOSUniversalBinary.py", line 298, in build
    subprocess.check_call(["cmake", "--build", ".",
  File "/opt/homebrew/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--parallel', '8']' returned non-zero exit status 2.

[ build ]                                                                             ⍉
>
```
i've installed qt5 from brew (`brew install qt@5`)
i do not have python2, and available python3 seems to match with `../BuildMacOSUniversalBinary.py` which have header:
```
!/usr/bin/env python3
```
both build seems to fail at moltenvk build step.

since it has been reopened  i would like to add, i was able to pass that moltenvk error, and also those `error: 'path' is unavailable: introduced in macOS 10.15`.
i added this to the cmakelist.txt file:  
```
# set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14.0" CACHE STRING "")

  set (CMAKE_OSX_DEPLOYMENT_TARGET "12.0" CACHE STRING "Minimum OS X deployment version" FORCE)
  set (CMAKE_OSX_ARCHITECTURES "x86_64" CACHE STRING "Architectures" FORCE)
  set (CMAKE_XCODE_ATTRIBUTE_MACOSX_DEPLOYMENT_TARGET[arch=x86_64] "12.0" CACHE STRING "arm 64 minimum deployment target" FORCE)
```
i also tried to change from  x86_64 with arm64. result is always the same.
it compile up to 100% then fails on final step:

```
[100%] Linking CXX executable ../../../Binaries/Dolphin.app/Contents/MacOS/Dolphin
Undefined symbols for architecture x86_64:
  "_win_w", referenced from:
      RenderWidget::event(QEvent*) in RenderWidget.cpp.o
  "_win_h", referenced from:
      RenderWidget::event(QEvent*) in RenderWidget.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/build.make:11186: Binaries/Dolphin.app/Contents/MacOS/Dolphin] Error 1
make[1]: *** [CMakeFiles/Makefile2:2546: Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```
i tried full stack x86_84 (using rosetta2 for the terminal, then x86_64 homebrew install, and qt5.12.12 from official site).
i also tried full stack arm64 (arm64 homebrew for llvm, python, cmale, make ..)
in both cases i end up there with those:` Undefined symbols for architecture x86_64 / amr64:   "_win_w",   "_win_h"`
maybe i should try with qt6 now.

@SirMangler 
thank you for your message, that's nice because i wasn't able to  building qt6 from sources (not enough memory error, process took 44gb and macos refused to continue) so i was out of options and ideas.
i modified Source/Core/DolphinQt/RenderWidget.cpp as you told, i also did `make clean` and `rm -fr ./*` in build directory, and i ran again `cmake ..` then `make -j5` but unfortunately the end result was the same.
```
[100%] Linking CXX executable ../../../Binaries/Dolphin.app/Contents/MacOS/Dolphin
Undefined symbols for architecture x86_64:
  "_win_w", referenced from:
      RenderWidget::event(QEvent*) in RenderWidget.cpp.o
  "_win_h", referenced from:
      RenderWidget::event(QEvent*) in RenderWidget.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [Binaries/Dolphin.app/Contents/MacOS/Dolphin] Error 1
make[1]: *** [Source/Core/DolphinQt/CMakeFiles/dolphin-emu.dir/all] Error 2
make: *** [all] Error 2

[ build ]                                                                     ⍉
> 
```
here's my RenderWidget.cpp:  (from line 454 to 471)
```
  case QEvent::Resize:
  {
    #if defined(CIFACE_USE_WIN32) || defined(CIFACE_USE_XLIB) 
    const QResizeEvent* se = static_cast<QResizeEvent*>(event);
    QSize new_size = se->size();

    QScreen* screen = window()->windowHandle()->screen();

    const auto dpr = screen->devicePixelRatio();

    // Window Centre for PrimeHack mouselock.
    win_w = (new_size.width() * dpr) / 2;
    win_h = (new_size.height() * dpr) / 2;

    emit SizeChanged(new_size.width() * dpr, new_size.height() * dpr);
    #endif
    break;
  }
```
but unfortunately this wasn't enough. 
however, i noticed two lines above (451-452):
https://github.com/shiiion/dolphin/blob/f82de185aa2365eb34b6567e67d37e3ae40f0fb4/Source/Core/DolphinQt/RenderWidget.cpp#L451-L452
```
    win_w = size().width() / 2;
    win_h = size().height() / 2;
```
so possibly there's still more references to correct for this problem.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ cmake_minimum_required(VERSION 3.10)
 # This is done in the universal binary building script to build a binary that 
 # runs on 10.13 on x86_64 computers, while still containing an arm64 slice.
 
- set(CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE STRING "")
+set(CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE STRING "")
 
 set(CMAKE_USER_MAKE_RULES_OVERRIDE "CMake/FlagsOverride.cmake")
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ cmake_minimum_required(VERSION 3.10)
 # This is done in the universal binary building script to build a binary that 
 # runs on 10.13 on x86_64 computers, while still containing an arm64 slice.
 
-set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14.0" CACHE STRING "")
+ set(CMAKE_OSX_DEPLOYMENT_TARGET "12.0" CACHE STRING "")
 
 set(CMAKE_USER_MAKE_RULES_OVERRIDE "CMake/FlagsOverride.cmake")
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ cmake_minimum_required(VERSION 3.10)
 # This is done in the universal binary building script to build a binary that 
 # runs on 10.13 on x86_64 computers, while still containing an arm64 slice.
 
- set(CMAKE_OSX_DEPLOYMENT_TARGET "12.0" CACHE STRING "")
+ set(CMAKE_OSX_DEPLOYMENT_TARGET "11.0" CACHE STRING "")
 
 set(CMAKE_USER_MAKE_RULES_OVERRIDE "CMake/FlagsOverride.cmake")
 

--- a/Externals/MoltenVK/patches/fix-shared-buffer-view.patch
+++ b/Externals/MoltenVK/patches/fix-shared-buffer-view.patch
@@ -1,0 +1,21 @@
+diff --git a/MoltenVK/MoltenVK/GPUObjects/MVKBuffer.mm b/MoltenVK/MoltenVK/GPUObjects/MVKBuffer.mm
+index 7975f72b..6239b967 100644
+--- a/MoltenVK/MoltenVK/GPUObjects/MVKBuffer.mm
++++ b/MoltenVK/MoltenVK/GPUObjects/MVKBuffer.mm
+@@ -311,7 +311,16 @@
+                                                                             width: _textureSize.width
+                                                                            height: _textureSize.height
+                                                                         mipmapped: NO];
++#if MVK_MACOS
++        // Textures on Mac cannot use shared storage, so force managed.
++        if (_buffer->getMTLBuffer().storageMode == MTLStorageModeShared) {
++            mtlTexDesc.storageMode = MTLStorageModeManaged;
++        } else {
++            mtlTexDesc.storageMode = _buffer->getMTLBuffer().storageMode;
++        }
++#else
+             mtlTexDesc.storageMode = mtlBuff.storageMode;
++#endif
+             mtlTexDesc.cpuCacheMode = mtlBuff.cpuCacheMode;
+             mtlTexDesc.usage = usage;
+         }

--- a/Source/Core/DolphinQt/CMakeLists.txt
+++ b/Source/Core/DolphinQt/CMakeLists.txt
@@ -558,9 +558,11 @@ if(APPLE)
     set_source_files_properties("${SOURCE_DIR}/Package/Release/MoltenVK/dylib/macOS/libMoltenVK.dylib" PROPERTIES MACOSX_PACKAGE_LOCATION Frameworks GENERATED ON)
   endif()
 
-  # Update library references to make the bundle portable
-  include(DolphinPostprocessBundle)
-  dolphin_postprocess_bundle(dolphin-emu)
+  if(NOT APPLE)
+    # Update library references to make the bundle portable
+    include(DolphinPostprocessBundle)
+    dolphin_postprocess_bundle(dolphin-emu)
+ endif()
   # Fix rpath
   add_custom_command(TARGET dolphin-emu
     POST_BUILD COMMAND

--- a/Source/Core/DolphinQt/CMakeLists.txt
+++ b/Source/Core/DolphinQt/CMakeLists.txt
@@ -562,7 +562,7 @@ if(APPLE)
     # Update library references to make the bundle portable
     include(DolphinPostprocessBundle)
     dolphin_postprocess_bundle(dolphin-emu)
- endif()
+  endif()
   # Fix rpath
   add_custom_command(TARGET dolphin-emu
     POST_BUILD COMMAND

--- a/Source/Core/DolphinQt/RenderWidget.cpp
+++ b/Source/Core/DolphinQt/RenderWidget.cpp
@@ -447,7 +447,7 @@ bool RenderWidget::event(QEvent* event)
     emit FocusChanged(false);
     break;
   case QEvent::Move:
-#if defined(CIFACE_USE_WIN32) || defined(CIFACE_USE_XLIB) 
+#if defined(CIFACE_USE_WIN32) || defined(CIFACE_USE_XLIB) || defined(CIFACE_USE_OSX)
     SetCursorLocked(m_cursor_locked);
     win_w = size().width() / 2;
     win_h = size().height() / 2;
@@ -456,7 +456,7 @@ bool RenderWidget::event(QEvent* event)
     break;
   case QEvent::Resize:
   {
-#if defined(CIFACE_USE_WIN32) || defined(CIFACE_USE_XLIB) 
+#if defined(CIFACE_USE_WIN32) || defined(CIFACE_USE_XLIB) || defined(CIFACE_USE_OSX)
     const QResizeEvent* se = static_cast<QResizeEvent*>(event);
     QSize new_size = se->size();
 

--- a/Source/Core/InputCommon/CMakeLists.txt
+++ b/Source/Core/InputCommon/CMakeLists.txt
@@ -111,6 +111,8 @@ if(WIN32)
   )
 elseif(APPLE)
   target_sources(inputcommon PRIVATE
+    QuartzInputMouse.h
+    QuartzInputMouse.mm
     ControllerInterface/OSX/OSX.h
     ControllerInterface/OSX/OSX.mm
     ControllerInterface/OSX/OSXJoystick.h

--- a/Source/Core/InputCommon/ControllerEmu/ControlGroup/PrimeHackModes.cpp
+++ b/Source/Core/InputCommon/ControllerEmu/ControlGroup/PrimeHackModes.cpp
@@ -21,7 +21,7 @@ namespace ControllerEmu
   // Check to see if we're on a platform with a mouse class.
   bool PrimeHackModes::GetMouseSupported() const
   {
-#if defined CIFACE_USE_WIN32 || defined CIFACE_USE_XLIB
+#if defined CIFACE_USE_WIN32 || defined CIFACE_USE_XLIB || defined CIFACE_USE_OSX
     return true;
 #else
     return false;

--- a/Source/Core/InputCommon/ControllerInterface/Quartz/QuartzKeyboardAndMouse.mm
+++ b/Source/Core/InputCommon/ControllerInterface/Quartz/QuartzKeyboardAndMouse.mm
@@ -9,6 +9,7 @@
 #include <Cocoa/Cocoa.h>
 
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "InputCommon/QuartzInputMouse.h"
 
 namespace ciface::Quartz
 {
@@ -141,7 +142,7 @@ KeyboardAndMouse::KeyboardAndMouse(void* view)
   // keys that aren't being recognized, bump this number up!
   for (int keycode = 0; keycode < 0x80; ++keycode)
     AddInput(new Key(keycode));
-
+  
   // Add combined left/right modifiers with consistent naming across platforms.
   AddCombinedInput("Alt", {"Left Alt", "Right Alt"});
   AddCombinedInput("Shift", {"Left Shift", "Right Shift"});
@@ -161,6 +162,7 @@ KeyboardAndMouse::KeyboardAndMouse(void* view)
       m_windowid = [[cocoa_view window] windowNumber];
     });
   }
+  prime::InitCocoaInputMouse(&m_windowid);
 
   // cursor, with a hax for-loop
   for (unsigned int i = 0; i < 4; ++i)
@@ -175,6 +177,7 @@ void KeyboardAndMouse::UpdateInput()
 {
   CGRect bounds = CGRectZero;
   CGWindowID windowid[1] = {m_windowid};
+  
   CFArrayRef windowArray = CFArrayCreate(nullptr, (const void**)windowid, 1, nullptr);
   CFArrayRef windowDescriptions = CGWindowListCreateDescriptionFromArray(windowArray);
   CFDictionaryRef windowDescription =

--- a/Source/Core/InputCommon/ControllerInterface/Quartz/QuartzKeyboardAndMouse.mm
+++ b/Source/Core/InputCommon/ControllerInterface/Quartz/QuartzKeyboardAndMouse.mm
@@ -142,7 +142,7 @@ KeyboardAndMouse::KeyboardAndMouse(void* view)
   // keys that aren't being recognized, bump this number up!
   for (int keycode = 0; keycode < 0x80; ++keycode)
     AddInput(new Key(keycode));
-  
+
   // Add combined left/right modifiers with consistent naming across platforms.
   AddCombinedInput("Alt", {"Left Alt", "Right Alt"});
   AddCombinedInput("Shift", {"Left Shift", "Right Shift"});
@@ -162,7 +162,7 @@ KeyboardAndMouse::KeyboardAndMouse(void* view)
       m_windowid = [[cocoa_view window] windowNumber];
     });
   }
-  prime::InitCocoaInputMouse(&m_windowid);
+  prime::InitQuartzInputMouse(&m_windowid);
 
   // cursor, with a hax for-loop
   for (unsigned int i = 0; i < 4; ++i)
@@ -177,7 +177,6 @@ void KeyboardAndMouse::UpdateInput()
 {
   CGRect bounds = CGRectZero;
   CGWindowID windowid[1] = {m_windowid};
-  
   CFArrayRef windowArray = CFArrayCreate(nullptr, (const void**)windowid, 1, nullptr);
   CFArrayRef windowDescriptions = CGWindowListCreateDescriptionFromArray(windowArray);
   CFDictionaryRef windowDescription =

--- a/Source/Core/InputCommon/QuartzInputMouse.h
+++ b/Source/Core/InputCommon/QuartzInputMouse.h
@@ -4,7 +4,7 @@
 namespace prime
 {
 
-bool InitCocoaInputMouse(uint32_t* windowid);
+bool InitQuartzInputMouse(uint32_t* windowid);
 
 class QuartzInputMouse: public GenericMouse
 {

--- a/Source/Core/InputCommon/QuartzInputMouse.h
+++ b/Source/Core/InputCommon/QuartzInputMouse.h
@@ -5,20 +5,20 @@ namespace prime
 {
 
 bool InitCocoaInputMouse(uint32_t* windowid);
-CGRect getBounds(uint32_t* m_windowid);
-CGPoint getWindowCenter(uint32_t* m_windowid);
 
-class CocoaInputMouse: public GenericMouse
+class QuartzInputMouse: public GenericMouse
 {
 public:
-  explicit CocoaInputMouse(uint32_t* windowid);
+  explicit QuartzInputMouse(uint32_t* windowid);
   void UpdateInput() override;
   void LockCursorToGameWindow() override;
 
 private:
   CGPoint current_loc, center;
-  CGEventRef event;
+  CGEventRef event{};
   uint32_t* m_windowid;
+  CGRect getBounds();
+  CGPoint getWindowCenter();
 };
 
 }

--- a/Source/Core/InputCommon/QuartzInputMouse.h
+++ b/Source/Core/InputCommon/QuartzInputMouse.h
@@ -11,15 +11,13 @@ CGPoint getWindowCenter(uint32_t* m_windowid);
 class CocoaInputMouse: public GenericMouse
 {
 public:
-  CocoaInputMouse(uint32_t* windowid);
+  explicit CocoaInputMouse(uint32_t* windowid);
   void UpdateInput() override;
   void LockCursorToGameWindow() override;
 
 private:
-  bool isFullScreen();
-  CGPoint current_loc, last_loc, origin;
+  CGPoint current_loc, center;
   CGEventRef event;
-  static const uint8 WINDOW_CHROME_OFFSET = 13;
   uint32_t* m_windowid;
 };
 

--- a/Source/Core/InputCommon/QuartzInputMouse.h
+++ b/Source/Core/InputCommon/QuartzInputMouse.h
@@ -1,0 +1,26 @@
+#include <ApplicationServices/ApplicationServices.h>
+#include "GenericMouse.h"
+
+namespace prime
+{
+
+bool InitCocoaInputMouse(uint32_t* windowid);
+CGRect getBounds(uint32_t* m_windowid);
+CGPoint getWindowCenter(uint32_t* m_windowid);
+
+class CocoaInputMouse: public GenericMouse
+{
+public:
+  CocoaInputMouse(uint32_t* windowid);
+  void UpdateInput() override;
+  void LockCursorToGameWindow() override;
+
+private:
+  bool isFullScreen();
+  CGPoint current_loc, last_loc, origin;
+  CGEventRef event;
+  static const uint8 WINDOW_CHROME_OFFSET = 13;
+  uint32_t* m_windowid;
+};
+
+}

--- a/Source/Core/InputCommon/QuartzInputMouse.mm
+++ b/Source/Core/InputCommon/QuartzInputMouse.mm
@@ -1,6 +1,3 @@
-#include <iostream>
-#include <math.h>
-
 #include "QuartzInputMouse.h"
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
 #include "Core/Host.h"

--- a/Source/Core/InputCommon/QuartzInputMouse.mm
+++ b/Source/Core/InputCommon/QuartzInputMouse.mm
@@ -1,6 +1,12 @@
 #include "QuartzInputMouse.h"
-#include "InputCommon/ControllerInterface/ControllerInterface.h"
 #include "Core/Host.h"
+
+/**
+ * This interface works by centering the cursor within the window
+ * ever time the `UpdateInput` method is called. It then calculates
+ * the delta based on how far the mouse has moved from the center of
+ * the screen before it is re-centered.
+ */
 
 
 int win_w = 0, win_h = 0;
@@ -17,7 +23,7 @@ bool InitCocoaInputMouse(uint32_t* windowid)
 CocoaInputMouse::CocoaInputMouse(uint32_t* windowid)
 {
   m_windowid = windowid;
-  origin = last_loc = getWindowCenter(m_windowid);
+  center = getWindowCenter(m_windowid);
 }
 
 void CocoaInputMouse::UpdateInput() 
@@ -25,44 +31,32 @@ void CocoaInputMouse::UpdateInput()
   event = CGEventCreate(nil);
   current_loc = CGEventGetLocation(event);
   CFRelease(event);
-  origin = getWindowCenter(m_windowid);
-
-  if (cursor_locked)
+  center = getWindowCenter(m_windowid);
+  if (Host_RendererHasFocus() && cursor_locked)
   {
-      this->dx += current_loc.x - origin.x;
-      this->dy += current_loc.y - origin.y;
-      if (!isFullScreen()) {
-        this->dy -= WINDOW_CHROME_OFFSET;
-      }
-      last_loc = current_loc;
+    this->dx += current_loc.x - center.x;
+    this->dy += current_loc.y - center.y;
   }
   LockCursorToGameWindow();
-
 }
 
 void CocoaInputMouse::LockCursorToGameWindow()
 {
   if (Host_RendererHasFocus() && cursor_locked)
   {
-    Host_RendererUpdateCursor(true);
-    CGDisplayHideCursor(CGMainDisplayID());
+    // Hack to avoid short bit of input supression after warp
+    // Credit/explanation: https://stackoverflow.com/a/17559012/7341382
+      CGWarpMouseCursorPosition(center);
+      CGAssociateMouseAndMouseCursorPosition(true);
+      CGDisplayHideCursor(CGMainDisplayID());
   }
   else
   {
     cursor_locked = false;
     Host_RendererUpdateCursor(false);
+    CGAssociateMouseAndMouseCursorPosition(true);
     CGDisplayShowCursor(CGMainDisplayID());
   }
-}
-
-bool CocoaInputMouse::isFullScreen()
-{
-  // TODO: Right now the game must be played on the main display.
-  auto bounds = getBounds(m_windowid);
-  auto mainDisplayId = CGMainDisplayID();
-  auto width = CGDisplayPixelsWide(mainDisplayId);
-  auto height = CGDisplayPixelsHigh(mainDisplayId);
-  return (bounds.size.width == width && bounds.size.height == height);
 }
 
 CGRect getBounds(uint32_t* m_windowid)

--- a/Source/Core/InputCommon/QuartzInputMouse.mm
+++ b/Source/Core/InputCommon/QuartzInputMouse.mm
@@ -14,7 +14,7 @@ int win_w = 0, win_h = 0;
 namespace prime
 {
 
-bool InitCocoaInputMouse(uint32_t* windowid)
+bool InitQuartzInputMouse(uint32_t* windowid)
 {
   g_mouse_input = new QuartzInputMouse(windowid);
   return true;

--- a/Source/Core/InputCommon/QuartzInputMouse.mm
+++ b/Source/Core/InputCommon/QuartzInputMouse.mm
@@ -1,0 +1,104 @@
+#include <iostream>
+#include <math.h>
+
+#include "QuartzInputMouse.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "Core/Host.h"
+
+
+int win_w = 0, win_h = 0;
+
+namespace prime
+{
+
+bool InitCocoaInputMouse(uint32_t* windowid)
+{
+  g_mouse_input = new CocoaInputMouse(windowid);
+  return true;
+}
+
+CocoaInputMouse::CocoaInputMouse(uint32_t* windowid)
+{
+  m_windowid = windowid;
+  origin = last_loc = getWindowCenter(m_windowid);
+}
+
+void CocoaInputMouse::UpdateInput() 
+{
+  event = CGEventCreate(nil);
+  current_loc = CGEventGetLocation(event);
+  CFRelease(event);
+  origin = getWindowCenter(m_windowid);
+
+  if (cursor_locked)
+  {
+      this->dx += current_loc.x - origin.x;
+      this->dy += current_loc.y - origin.y;
+      if (!isFullScreen()) {
+        this->dy -= WINDOW_CHROME_OFFSET;
+      }
+      last_loc = current_loc;
+  }
+  LockCursorToGameWindow();
+
+}
+
+void CocoaInputMouse::LockCursorToGameWindow()
+{
+  if (Host_RendererHasFocus() && cursor_locked)
+  {
+    Host_RendererUpdateCursor(true);
+    CGDisplayHideCursor(CGMainDisplayID());
+  }
+  else
+  {
+    cursor_locked = false;
+    Host_RendererUpdateCursor(false);
+    CGDisplayShowCursor(CGMainDisplayID());
+  }
+}
+
+bool CocoaInputMouse::isFullScreen()
+{
+  // TODO: Right now the game must be played on the main display.
+  auto bounds = getBounds(m_windowid);
+  auto mainDisplayId = CGMainDisplayID();
+  auto width = CGDisplayPixelsWide(mainDisplayId);
+  auto height = CGDisplayPixelsHigh(mainDisplayId);
+  return (bounds.size.width == width && bounds.size.height == height);
+}
+
+CGRect getBounds(uint32_t* m_windowid)
+{
+  CGRect bounds = CGRectZero;
+  CGWindowID windowid[1] = {*m_windowid};
+  
+  CFArrayRef windowArray = CFArrayCreate(nullptr, (const void**)windowid, 1, nullptr);
+  CFArrayRef windowDescriptions = CGWindowListCreateDescriptionFromArray(windowArray);
+  CFDictionaryRef windowDescription =
+      static_cast<CFDictionaryRef>(CFArrayGetValueAtIndex(windowDescriptions, 0));
+
+  if (CFDictionaryContainsKey(windowDescription, kCGWindowBounds))
+  {
+    CFDictionaryRef boundsDictionary =
+        static_cast<CFDictionaryRef>(CFDictionaryGetValue(windowDescription, kCGWindowBounds));
+
+    if (boundsDictionary != nullptr)
+      CGRectMakeWithDictionaryRepresentation(boundsDictionary, &bounds);
+  }
+
+  CFRelease(windowDescriptions);
+  CFRelease(windowArray);
+  return bounds;
+}
+
+CGPoint getWindowCenter(uint32_t* m_windowid)
+{
+  auto bounds = getBounds(m_windowid);
+  double x = bounds.origin.x + (bounds.size.width / 2);
+  double y = bounds.origin.y + (bounds.size.height / 2);
+  CGPoint center = {x, y};
+  return center;
+}
+
+}

--- a/Source/Core/InputCommon/QuartzInputMouse.mm
+++ b/Source/Core/InputCommon/QuartzInputMouse.mm
@@ -16,22 +16,22 @@ namespace prime
 
 bool InitCocoaInputMouse(uint32_t* windowid)
 {
-  g_mouse_input = new CocoaInputMouse(windowid);
+  g_mouse_input = new QuartzInputMouse(windowid);
   return true;
 }
 
-CocoaInputMouse::CocoaInputMouse(uint32_t* windowid)
+QuartzInputMouse::QuartzInputMouse(uint32_t* windowid)
 {
   m_windowid = windowid;
-  center = getWindowCenter(m_windowid);
+  center = current_loc = getWindowCenter();
 }
 
-void CocoaInputMouse::UpdateInput() 
+void QuartzInputMouse::UpdateInput()
 {
   event = CGEventCreate(nil);
   current_loc = CGEventGetLocation(event);
   CFRelease(event);
-  center = getWindowCenter(m_windowid);
+  center = getWindowCenter();
   if (Host_RendererHasFocus() && cursor_locked)
   {
     this->dx += current_loc.x - center.x;
@@ -40,11 +40,11 @@ void CocoaInputMouse::UpdateInput()
   LockCursorToGameWindow();
 }
 
-void CocoaInputMouse::LockCursorToGameWindow()
+void QuartzInputMouse::LockCursorToGameWindow()
 {
   if (Host_RendererHasFocus() && cursor_locked)
   {
-    // Hack to avoid short bit of input supression after warp
+    // Hack to avoid short bit of input suppression after warp
     // Credit/explanation: https://stackoverflow.com/a/17559012/7341382
       CGWarpMouseCursorPosition(center);
       CGAssociateMouseAndMouseCursorPosition(true);
@@ -59,7 +59,7 @@ void CocoaInputMouse::LockCursorToGameWindow()
   }
 }
 
-CGRect getBounds(uint32_t* m_windowid)
+CGRect QuartzInputMouse::getBounds()
 {
   CGRect bounds = CGRectZero;
   CGWindowID windowid[1] = {*m_windowid};
@@ -83,13 +83,12 @@ CGRect getBounds(uint32_t* m_windowid)
   return bounds;
 }
 
-CGPoint getWindowCenter(uint32_t* m_windowid)
+CGPoint QuartzInputMouse::getWindowCenter()
 {
-  auto bounds = getBounds(m_windowid);
-  double x = bounds.origin.x + (bounds.size.width / 2);
-  double y = bounds.origin.y + (bounds.size.height / 2);
-  CGPoint center = {x, y};
-  return center;
+  const auto bounds = getBounds();
+  const double x = bounds.origin.x + (bounds.size.width / 2);
+  const double y = bounds.origin.y + (bounds.size.height / 2);
+  return CGPointMake(x, y);
 }
 
 }

